### DISCUSSION TITLE
get store by id

### DISCUSSION THREAD

**Discussion by tyrunasj** [vuejs/pinia#753] [Help and Questions] [answered]
Hi devs, 

Maybe someone can answer - is there a possibility to access the "pinia" store by id. 
I mean I can create a number of stores as composables like useSomeStore. 
But what if my component needs to access different stores depending on a variable that is provided eg. through the props ?

Thank you in advance TJ

**Comment by posva** [comment, marked as answer]
You can't do it (at least with public API) because by design you should import every store you use so code splitting works correctly. In order to make what you want work, you would have to keep track of installed stores. This is actually very simple to setup with a plugin:

```ts
import type { StoreGeneric } from 'pinia'

const storeMap = new Map<string, StoreGeneric>()
pinia.use(({ store }) => {
  storeMap.set(store.$id, store)
})
```

Then you can call `storeMap.get(id)` but it won't be able to instantiate the store because it's not aware of all the imports. You could create another function that dynamically imports the store based on its filename if it hasn't been used yet but that would make it async, forcing you to use Suspense as well.

**Reply by gllinzbk1** [reply]
@posva in nuxt3 ，how to do this plugin, how to work
 '
pinia.use(({ store }) => {
  storeMap.set(store.$id, store)
})
'

**Comment by tyrunasj** [comment]
Eduardo thanks for the explanation.
